### ISSUE TITLE
Prophet + lightgbm unused argument error

### ISSUE THREAD

**Issue by sdbentrup** [AlbertoAlmuinha/boostime#8]
Hello, I am getting the following error:

Error in lightgbm::lgb.Dataset(data = prepare_df_lgbm(predictors), label = prophet_residuals,  : 
  unused Argument (feature_pre_filter = FALSE)

It seems the argument ``` feature_pre_filter ``` maybe is not used in lgb.Dataset() in R.

Reprex:
```r
library(tidymodels)
library(boostime)
library(modeltime)
library(tidyverse)
library(timetk)
library(lubridate)
# This toggles plots from plotly (interactive) to ggplot (static)

m750 <- m4_monthly %>% filter(id == "M750")

splits <- initial_time_split(m750, prop = 0.8)

model_prophet_lightgbm <- boost_prophet() %>%
  set_engine("prophet_lightgbm") %>%
  fit(value ~ date + month(date), data = training(splits))
``` 

I am happy to help test if needed :)

session info:
<html>
<body>

R version 4.5.0 (2025-04-11 ucrt) Platform: x86_64-w64-mingw32/x64 Running under: Windows 11 x64 (build 26100)  Matrix products: default   LAPACK version 3.12.1  locale: [1] LC_COLLATE=German_Austria.utf8  LC_CTYPE=German_Austria.utf8    LC_MONETARY=German_Austria.utf8 [4] LC_NUMERIC=C                    LC_TIME=German_Austria.utf8      time zone: Europe/Berlin tzcode source: internal  attached base packages: [1] stats     graphics  grDevices utils     datasets  methods   base       other attached packages:  [1] rvest_1.0.4                ggthemes_5.1.0             shapviz_0.9.7               [4] rstatix_0.7.2              plotly_4.10.4              data.table_1.17.4           [7] PortfolioAnalytics_2.1.0   baguette_1.1.0             bonsai_0.3.2               [10] timetk_2.9.0               lubridate_1.9.4            forcats_1.0.0              [13] stringr_1.5.1              readr_2.1.5                tidyverse_2.0.0            [16] PerformanceAnalytics_2.0.8 quantmod_0.4.27            TTR_0.24.4                 [19] xts_0.14.1                 zoo_1.8-14                 tidyquant_1.0.11           [22] doFuture_1.1.1             foreach_1.5.2              future.apply_1.20.0        [25] future_1.58.0              tictoc_1.2.1               prophet_1.0                [28] rlang_1.1.6                Rcpp_1.0.14                lightgbm_4.6.0             [31] xgboost_1.7.11.1           finetune_1.2.1             boostime_0.1.0             [34] modeltime.ensemble_1.0.4   modeltime.resample_0.2.3   modeltime_1.3.1            [37] yardstick_1.3.2            workflowsets_1.1.1         workflows_1.2.0            [40] tune_1.3.0                 tidyr_1.3.1                tibble_3.3.0               [43] rsample_1.3.0              recipes_1.3.1              purrr_1.0.4                [46] parsnip_1.3.2              modeldata_1.4.0            infer_1.0.8                [49] ggplot2_3.5.2              dplyr_1.1.4                dials_1.4.0                [52] scales_1.4.0               broom_1.0.8                tidymodels_1.3.0            loaded via a namespace (and not attached):   [1] splines_4.5.0             hardhat_1.4.1             janitor_2.2.1               [4] rpart_4.1.24              sparsevctrs_0.3.4         lifecycle_1.0.4             [7] Rdpack_2.6.4              StanHeaders_2.32.10       GenSA_1.1.14.1             [10] globals_0.18.0            lattice_0.22-6            MASS_7.3-65                [13] backports_1.5.0           magrittr_2.0.3            C50_0.2.0                  [16] pkgbuild_1.4.8            minqa_1.2.8               RColorBrewer_1.1-3         [19] abind_1.4-8               quadprog_1.5-8            nnet_7.3-20                [22] ipred_0.9-15              lava_1.8.1                pso_1.0.4                  [25] inline_0.3.21             listenv_0.9.1             parallelly_1.45.0          [28] codetools_0.2-20          xml2_1.3.8                tidyselect_1.2.1           [31] shape_1.4.6.1             farver_2.1.2              lme4_1.1-37                [34] matrixStats_1.5.0         stats4_4.5.0              Rsymphony_0.1-33           [37] jsonlite_2.0.0            Formula_1.2-5             survival_3.8-3             [40] iterators_1.0.14          tools_4.5.0               glue_1.8.0                 [43] prodlim_2025.04.28        gridExtra_2.3             loo_2.8.0                  [46] withr_3.0.2               numDeriv_2016.8-1.1       fastmap_1.2.0              [49] boot_1.3-31               digest_0.6.37             timechange_0.3.0           [52] R6_2.6.1                  inum_1.0-5                utf8_1.2.6                 [55] generics_0.1.4            class_7.3-23              httr_1.4.7                 [58] htmlwidgets_1.6.4         pkgconfig_2.0.3           gtable_0.3.6               [61] timeDate_4041.110         registry_0.5-1            GPfit_1.0-9                [64] furrr_0.3.1               htmltools_0.5.8.1         sfd_0.1.0                  [67] carData_3.0-5             gower_1.0.2               reformulas_0.4.1           [70] snakecase_0.11.1          rstudioapi_0.17.1         ROI_1.0-1                  [73] tzdb_0.5.0                reshape2_1.4.4            checkmate_2.3.2            [76] nlme_3.1-168              curl_6.2.2                nloptr_2.2.1               [79] parallel_4.5.0            libcoin_1.0-10            warp_0.2.1                 [82] pillar_1.10.2             grid_4.5.0                vctrs_0.6.5                [85] slam_0.1-55               car_3.1-3                 lhs_1.2.0                  [88] RobStatTM_1.0.11          Cubist_0.5.0              mvtnorm_1.3-3              [91] cli_3.6.5                 compiler_4.5.0            crayon_1.5.3               [94] labeling_0.4.3            mco_1.17                  plyr_1.8.9                 [97] stringi_1.8.7             rstan_2.32.7              viridisLite_0.4.2         [100] QuickJSR_1.8.0            lazyeval_0.2.2            glmnet_4.1-9              [103] V8_6.0.4                  Matrix_1.7-3              catboost_1.2.8            [106] hms_1.1.3                 extraDistr_1.10.0         ROI.plugin.symphony_1.0-0 [109] rbibutils_2.3             slider_0.3.2              partykit_1.2-24           [112] RcppParallel_5.1.10       DiceDesign_1.10
--
 
  |  
 

<br class="Apple-interchange-newline">
</body>
</html>
